### PR TITLE
[KB-264] Rewrite GBR

### PR DESCRIPTION
**Pull request by Pattrigue**

### CODE REVIEW

File: report_thesis/src/references.bib [hunk 1]
```diff
@@ -595,3 +595,10 @@ @misc{duan_ngboost_2020
 	keywords = {Computer Science - Machine Learning, Statistics - Machine Learning},
 	annote = {Comment: Accepted for ICML 2020},
 }
+
+@article{zemel_gbr,
+	title = {A {Gradient}-{Based} {Boosting} {Algorithm} for {Regression} {Problems}},
+	abstract = {In adaptive boosting, several weak learners trained sequentially are combined to boost the overall algorithm performance. Recently adaptive boosting methods for classification problems have been derived as gradient descent algorithms. This formulation justifies key elements and parameters in the methods, all chosen to optimize a single common objective function. We propose an analogous formulation for adaptive boosting of regression problems, utilizing a novel objective function that leads to a simple boosting algorithm. We prove that this method reduces training error, and compare its performance to other regression methods.},
+	language = {en},
+	author = {Zemel, Richard S and Pitassi, Toniann},
+}
```
Comment (chhoumann) [inline report_thesis/src/references.bib:604]: Wondering if it's better to use the original paper? [Greedy Function Approximation: A Gradient Boosting Machine](https://jerryfriedman.su.domains/ftp/trebst.pdf)

It's also the reference used for the sklearn implementation, which we use.

File: report_thesis/src/sections/background.tex [hunk 2]
```diff
@@ -226,28 +226,50 @@ \subsubsection{Extra Trees Regressor (ETR)}
 However, it often achieves better generalization performance, especially in high-dimensional or noisy datasets.
 
 \subsubsection{Gradient Boosting Regression (GBR)}\label{sec:gradientboost}
-In this section we introduce \gls{gbr} primarily based on \citet{James2023AnIS}.
-\gls{gbr} is an ensemble learning method that sequentially construct multiple decision trees, each one correcting the errors of the previous one, using gradient descent and boosting techniques.
-Each tree is small, with few terminal nodes, preventing large adjustments based on a single tree's predictions.
-This ensures that each tree makes small and simple error corrections, such that each step refines the model's performance more reliably.
+The following is an overview of the \gls{gbr} algorithm based on \citet{zemel_gbr}.
```
Comment (chhoumann) [inline report_thesis/src/sections/background.tex:229]: This cite doesn't have anything to do with us nor the equations we present in this section.

It's important that, if we cite something, that we _actually_ used that paper. The equations we have here are not found in that paper. Can't add a paper retrospectively.

File: report_thesis/src/references.bib (hunk 1 above)
Comment (chhoumann) [reply report_thesis/src/references.bib:604]: And if you really don't want to read the original work, use, e.g., section 10.10.2 in the great Hastie's [Elements of Statistical Learning](https://hastie.su.domains/Papers/ESLII.pdf). 

IIRC, GBRT minimizes squared error via $\frac{1}{2}\[y_i-f(x_i)\]^2$, which doesn't necessitate calculating gradients, just the residuals $y_i-f(x_i)$.

**Review by chhoumann — CHANGES_REQUESTED** [review]
We really need to get the source in order. Can't write a section saying we based in on a work, but then use nothing from that work.

I think a rewrite using a proper source, from scratch, is most appropriate.

File: report_thesis/src/sections/background.tex [hunk 3]
```diff
@@ -226,28 +226,40 @@ \subsubsection{Extra Trees Regressor (ETR)}
 However, it often achieves better generalization performance, especially in high-dimensional or noisy datasets.
 
 \subsubsection{Gradient Boosting Regression (GBR)}\label{sec:gradientboost}
-In this section we introduce \gls{gbr} primarily based on \citet{James2023AnIS}.
-\gls{gbr} is an ensemble learning method that sequentially construct multiple decision trees, each one correcting the errors of the previous one, using gradient descent and boosting techniques.
-Each tree is small, with few terminal nodes, preventing large adjustments based on a single tree's predictions.
-This ensures that each tree makes small and simple error corrections, such that each step refines the model's performance more reliably.
+Continuing with the discussion of ensemble learning models, we introduce \gls{gbr} based on Chapter 10.10 of \citet{hastie_elements}.
```
Comment (Ivikhostrup) [inline report_thesis/src/sections/background.tex:229]: I might be wrong, but isnt it typical to include the relevant pages in the cite? Meaning, we add these pages in the references, rather than specifying them in the text?

File: report_thesis/src/sections/background.tex [hunk 4]
```diff
@@ -226,28 +226,40 @@ \subsubsection{Extra Trees Regressor (ETR)}
...
+
+Gradient Boosting is a powerful machine learning technique used for regression and classification tasks.
+The core idea is to build a model from weak learners, typically decision trees, in a sequential manner.
+Each subsequent model aims to correct the errors of its predecessor, resulting in a strong predictive model.
+
+\gls{gbr} is a specific implementation of gradient boosting for regression tasks.
+\gls{gbr} builds a series of decision trees sequentially, where each tree tries to reduce the errors made by the previous one.
```
Comment (Ivikhostrup) [inline report_thesis/src/sections/background.tex:236]:
```suggestion
Gradient Boosting is a powerful machine learning technique used for regression and classification tasks, that builds strong predictive models from a sequence of weaker ones, typically decision trees.
The core idea of this method lies in its sequential approach, where each tree is constructed to address and minimize the residuals or prediction errors of its predecessor.
Within this framework, \gls{gbr} applies this principle specifically to regression tasks, strategically refining each subsequent model to enhance overall accuracy.
```

My goal with this suggestion is to avoid a bit of the repetition related to "correcting the errors of the previous one". 
What do you think about this idea? I thought, since gbr builds on this concept, this might provide a better flow.

**Review by chhoumann — CHANGES_REQUESTED** [review]

**Comment by chhoumann** [comment]
In the interest of (hopefully) finishing this part of the report, I've given this section a shot: https://github.com/chhoumann/thesis-chemcam/pull/173

**Comment by chhoumann** [comment]
Closing in favor of #173
